src/pymovements/dataset/dataset_definition.py

```diff
@@ -154,13 +238,11 @@ class DatasetDefinition:
 
     has_files: dict[str, bool] = field(default_factory=dict)
 
-    mirrors: dict[str, list[str]] | dict[str, tuple[str, ...]] = field(default_factory=dict)
+    mirrors: dict[str, Sequence[str]] = field(default_factory=dict)
 
-    resources: dict[str, list[dict[str, str]]] | dict[str, tuple[dict[str, str], ...]] = field(
-        default_factory=dict,
-    )
+    resources: ResourceDefinitions = field(default_factory=ResourceDefinitions)
```

Comment:
This PR makes sure that `DatasetDefinition.resources` is always of type `Resource`.

The `__init__()` is more permissive: it allows also `ResourceLike` and `None`, but it always initializes a `Resource` object from the input.

This way we make sure that we have a consistent format later on and avoid needing to take care of different data structures.